packages/astro/src/core/render/dev/index.ts

```diff
@@ -40,11 +40,19 @@ export type ComponentPreload = [SSRLoadedRenderer[], ComponentInstance];
 export type RenderResponse = { type: 'html'; html: string } | { type: 'response'; response: Response };
 
 const svelteStylesRE = /svelte\?svelte&type=style/;
+const rendererCache = new Map<string, SSRLoadedRenderer['ssr']>();
```

Comment:
Can you leave a comment here that this is required to prevent bugs? I removed it assuming it was no longer needed for perf, but had no idea it was running defense on bugs!

Reply:
Yeah, I'm curious if it ever _was_ running defense on bugs to be honest. It seemed like a performance choice to me as well. I'll add a note 👍

Reply:
Updated!